packages/quantic/force-app/main/default/lwc/quanticUtils/quanticUtils.js

```diff
@@ -419,3 +419,36 @@ export class RelativeDateFormatter {
     return I18nUtils.format(label, relativeDate.amount);
   }
 }
+
+export class Store {
+
+  static initialize() {
+    return { 
+      state : { 
+        facets: {},
+        numericFacets: {},
+        dateFacets: {},
+        timeFrameFacets: {},
+        categoryFacets: {},
+      }}
+  }
+  /**
+   * @param {Record<String, unknown>} store
+   * @param {string} facetType
+   * @param {{ label?: string; facetId: any; format?: Function;}} data
+   */
+  static registerFacetToStore(store, facetType, data) {
```

Comment:
I was :snooping: on that PR, and I don't think it's a good idea to access and modify the `state` object from redux like this.

Immutability is an important part of redux, and it allows a lot of internal optimization, plugins, debugging tools, etc etc to work properly.

https://redux.js.org/faq/immutable-data

Reading from the state (the `getFromStore` function) is fine, however.

If you really want to use Headless store (redux) to store/pass UI data between components, the correct way would be to use reducers.

If that seems too complex/overkill, then you could just add a static map like you are doing here (as long as you understand the consequences of having global mutable object), but not directly adding it to the `headless.store.state` object.

I also found this with a quick google search: https://www.lwc-redux.com/

That would also benefit re-rendering when a language or a specific label for a facet changes.
For example: Think about if there was a language selector option in an interface, or a way for admin to do live edit on the strings of their interface.

Reply:
I agree that it doesn't need to be in the headless store. And if we can work with a simple static map with no rendering triggers then I would also wait on using something like lwc-redux. Pretty cool though and thanks for your suggestion, @olamothe!